test/unit/TestRequirements.hs

```diff
@@ -0,0 +1,55 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Swarm requirements analysis tests
+module TestRequirements where
+
+import Control.Lens (view)
+import Data.Set qualified as S
+import Data.Text (Text)
+import Swarm.Language.Capability
+import Swarm.Language.Context qualified as Ctx
+import Swarm.Language.Pipeline
+import Swarm.Language.Requirement
+import Test.Tasty
+import Test.Tasty.HUnit
+import Witch (from)
+
+testRequirements :: TestTree
+testRequirements =
+  testGroup
+    "Requirements analysis"
+    [ testGroup
+        "Basic capabilities"
+        [ testCase "solar panel" $ "noop" `requiresCap` CPower
+        , testCase "move" $ "move" `requiresCap` CMove
+        , testCase "lambda" $ "\\x. x" `requiresCap` CLambda
+        , testCase "inl" $ "inl 3" `requiresCap` CSum
+        , testCase "cap from type" $ "inl () : rec t. Unit + t" `requiresCap` CRectype
+        ]
+    , testGroup
+        "Scope"
+        [ testCase "global var requirement does not apply to local var (#1914)" $
+            checkReqCtx
+              "def m = move end; def y = \\m. log (format m) end"
+              (maybe False ((CMove `S.notMember`) . capReqs) . Ctx.lookup "y")
+        ]
+    ]
+
+checkReqCtx :: Text -> (ReqCtx -> Bool) -> Assertion
+checkReqCtx code expect = check code (expect . view processedReqCtx)
+
+checkRequirements :: Text -> (Requirements -> Bool) -> Assertion
+checkRequirements code expect = check code (expect . view processedRequirements)
+
+requiresCap :: Text -> Capability -> Assertion
+requiresCap code cap = checkRequirements code ((cap `S.member`) . capReqs)
+
+check :: Text -> (ProcessedTerm -> Bool) -> Assertion
+check code expect = case processTerm code of
+  Left err -> assertFailure $ "Term processing failed: " ++ from @Text err
+  Right Nothing -> assertFailure "Term is empty"
+  Right (Just pt) -> assertBool "Predicate was false!" (expect pt)
```

Comment:
Is this not used anywhere else? I would expect this to be in `TestUtil`.

Reply:
Good idea, I'll move it there.